projeto_guiado_v2_williane.py

```diff
@@ -0,0 +1,187 @@
+        while True:
+            try:
+                presenca = input(f"Informar presença no dia #{i+1} (True/False): ").strip().lower()
+                if presenca not in ['true', 'false']:
+                    raise ValueError("Valor inválido. Insira 'True' ou 'False'.")
+                else:
+                    presenca_aluna.append(presenca == 'true')
+                    break
+            except ValueError as e:
+                print(e)
+    
+    # Coletando a nota de participação
+    while True:
+        try:
+            participacao = float(input("Insira a nota de participação da aluna: "))
+            if participacao < 0 or participacao > 10:
+                raise ValueError("Nota de participação fora do intervalo permitido (0-10).")
+            else:
+                break
+        except ValueError as e:
+            print(e)
+    
+    # Calculando o status de aprovação
+    media = sum(notas_aluna) / len(notas_aluna)
+    percent_presenca = presenca_aluna.count(True) / len(presenca_aluna) * 100
+    
+    if participacao >= 6 and percent_presenca >= 80 and media >= 6:
+        status = "Aprovada"
+    elif 3 <= media < 6:
+        status = "Recuperação"
+        fez_recuperacao = input("\nA média da aluna está baixa. Ela fez prova de recuperação? ").lower()
+        if fez_recuperacao == 'sim':
+            while True:
+                try:
+                    nota_recuperacao = float(input("Insira a nota de recuperação: "))
+                    if nota_recuperacao < 0 or nota_recuperacao > 10:
+                        raise ValueError("Nota de recuperação fora do intervalo permitido (0-10).")
+                    else:
+                        break
+                except ValueError as e:
+                    print(e)
+        else:
+            nota_recuperacao = None
+            print("Então essa aluna será reprovada por média.")
+    else:
+        status = "Reprovada"
+        nota_recuperacao = None
+    
+    # Adicionando a aluna ao dataset
+    chave = f"{nome} {sobrenome}"
+    dataset[chave] = {
+        "Turma": turma,
+        "Notas": notas_aluna,
+        "Presença": presenca_aluna,
+        "Nota de Participação": participacao,
+        "Status": status,
+        "Média Final": round(media, 2)  # Arredonda a média final para duas casas decimais
+    }
+    
+    # Exibindo o status de aprovação da aluna
+    print(f"\nStatus de aprovação da aluna '{chave}': {status}")
+    print(f"Média final: {media:.2f}")
+    if status == "Recuperação":
+        print(f"Nota de recuperação: {nota_recuperacao:.2f}")
+    
+    print(f"\nAluna '{chave}' adicionada com sucesso!")
+
+def consultar_lista_alunas():
+    if not dataset:
+        print("\nNão há alunas cadastradas.")
+    else:
+        print("\nLista de alunas cadastradas:")
+        for chave in dataset.keys():
+            print(chave)
+
+def consultar_faltas_aluna():
+    nome_aluna = input("\nDigite o nome completo da aluna para consultar as faltas: ")
```

Comment:
O dicionário possui uma tupla como chave. Aqui está uma sugestão de implementação simples:

**Sugestão:**
```suggestion
    nome = input('Informe o nome da aluna: ')
    sobrenome = input('Informe o sobrenome da aluna: ')
    nome_aluna = (nome, sobrenome)
```